spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/AggregateChange.java

```diff
@@ -199,6 +154,51 @@ static void setId(RelationalMappingContext context, RelationalConverter converte
 		return intermediateAccessor;
 	}
 
+	@SuppressWarnings("unchecked")
+	public void executeWith(Interpreter interpreter, RelationalMappingContext context, RelationalConverter converter) {
+
+		RelationalPersistentEntity<T> persistentEntity = entity != null
+				? (RelationalPersistentEntity<T>) context.getRequiredPersistentEntity(entity.getClass())
+				: null;
+
+		PersistentPropertyAccessor<T> propertyAccessor = //
+				persistentEntity != null //
+						? converter.getPropertyAccessor(persistentEntity, entity) //
+						: null;
+
+		actions.forEach(a -> {
+
+			a.executeWith(interpreter);
+
+			if (a instanceof DbAction.WithGeneratedId) {
+
+				Assert.notNull(persistentEntity,
```

Comment:
This code is deeply nested. How about extracting it to a class/method that makes the concept explicit?